scripts/ci/install_and_check_minimum_requirements.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -ex
+
+# minimum boost version
+BOOST_VERSION=1.83
+
+# minimum cmake version
+CMAKE_MAJOR=3
+CMAKE_MINOR=21
+CMAKE_PATCH=3
+CMAKE_FULL_VERSION="${CMAKE_MAJOR}.${CMAKE_MINOR}.${CMAKE_PATCH}"
+
+# minimum gcc/clang versions
+GCC_VERSION=13.3.0
+CLANG_VERSION=18.1.3
+
+# which compiler version to check in this script
+CHECK_GCC=false
+CHECK_CLANG=false
+
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --gcc)
+            CHECK_GCC=true
+            shift
+            ;;
+        --clang)
+            CHECK_CLANG=true
+            shift
+            ;;
+        *)
+            echo "Unknown option: $1"
+            echo "Usage: $0 [--gcc] [--clang]"
+            exit 1
+            ;;
+    esac
+done
+
+echo "-- Removing boost and CMake from the system"
+sudo apt-get -qq remove --purge 'libboost*' -y
+sudo apt-get -qq remove --purge cmake -y
+
+echo "-- Installing Boost ${BOOST_VERSION}"
+sudo apt-get -qq update -y
+sudo apt-get -qq install libboost${BOOST_VERSION}-all-dev -y
+INSTALLED_BOOST_VERSION=$(dpkg-query --showformat='${Version}' --show libboost${BOOST_VERSION}-all-dev 2>/dev/null || echo "none")
+if [[ "$INSTALLED_BOOST_VERSION" != ${BOOST_VERSION}* ]]; then
+    echo "Error: installed version of boost is $INSTALLED_BOOST_VERSION, expected $BOOST_VERSION"
+    exit 1
+fi
+
+echo "-- Installing CMake ${CMAKE_FULL_VERSION}"
+wget https://cmake.org/files/v${CMAKE_MAJOR}.${CMAKE_MINOR}/cmake-${CMAKE_FULL_VERSION}-linux-x86_64.tar.gz
+tar -xzf cmake-${CMAKE_FULL_VERSION}-linux-x86_64.tar.gz
+sudo mv cmake-${CMAKE_FULL_VERSION}-linux-x86_64 /opt/cmake-${CMAKE_FULL_VERSION}
+sudo ln -s /opt/cmake-${CMAKE_FULL_VERSION}/bin/* /usr/local/bin/
```

Comment:
Some quotes are missing. And please use long option names where available.

```suggestion
wget "https://cmake.org/files/v${CMAKE_MAJOR}.${CMAKE_MINOR}/cmake-${CMAKE_FULL_VERSION}-linux-x86_64.tar.gz"
tar --extract --gzip --file "cmake-${CMAKE_FULL_VERSION}-linux-x86_64.tar.gz"
sudo mv "cmake-${CMAKE_FULL_VERSION}-linux-x86_64" "/opt/cmake-${CMAKE_FULL_VERSION}"
sudo ln --symbolic "/opt/cmake-${CMAKE_FULL_VERSION}/bin/"* /usr/local/bin/
```

Reply:
Also with `apt-get` above, e.g. `apt-get --quiet=2 update --assume-yes`.

Also, note that man page says that quiet level 2 already implies `-y`.

Reply:
On the other hand in `[[ "$INSTALLED_BOOST_VERSION" != ${BOOST_VERSION}* ]]` the first quote is redundant.